signer/src/transaction_coordinator.rs

```diff
@@ -673,32 +763,39 @@ where
         // on the canonical Stacks blockchain.
         //
         // If the signers have already run DKG, then we know that all
-        // participating signers have completed it successfully (well, some
-        // may have failed suddenly at the end, and some may have dropped
-        // off during DKG, but yeah enough should have their DKG shares).
-        // So we can fall back on the stored DKG shares for getting the
-        // current aggregate key and associated signing set.
+        // participating signers should have the same view of the latest
+        // aggregate key, so we can fall back on the stored DKG shares for
+        // getting the current aggregate key and associated signing set.
         match db.get_last_key_rotation(bitcoin_chain_tip).await? {
             Some(last_key) => {
                 let aggregate_key = last_key.aggregate_key;
                 let signer_set = last_key.signer_set.into_iter().collect();
-                Ok((aggregate_key, signer_set))
-            }
-            None => {
-                let shares = db
-                    .get_last_encrypted_dkg_shares()
-                    .await?
-                    .ok_or(Error::MissingDkgShares)?;
-                let signer_set = shares.signer_set_public_keys.into_iter().collect();
-                Ok((shares.aggregate_key, signer_set))
+                Ok((Some(aggregate_key), signer_set))
             }
+            None => match db.get_last_encrypted_dkg_shares().await? {
+                Some(shares) => {
+                    let signer_set = shares.signer_set_public_keys.into_iter().collect();
+                    Ok((Some(shares.aggregate_key), signer_set))
+                }
+                None => Ok((None, self.context.config().signer.bootstrap_signing_set())),
```

Comment:
So if a signer joins the set after it's been bootstrapped, it will technically need to have the config updated to the current signer set pubkeys, right? Since it won't have any dkg shares/key rotations stored?

Reply:
Yup, that is correct.

A "new" signer needs to get the current signer set public keys. This seems like a definite problem for v2, but we have lots of problems like this for v2. Oh and it might be a problem for v1 as well, but that depends on what we need to deliver.

Reply:
Yeah okay I just wanted to make sure I was reading it right :) We have several options we've discussed so once we get the room/prio we can attack that problem